src/pages/resultsView/survival/SurvivalUtil.tsx

```diff
@@ -520,6 +531,80 @@ export function downSampling(
     });
 }
 
+export function floorScatterData(
+    scatterDataWithOpacity: ScatterData[]
+): ScatterData[] {
+    const eventInfoByMonth: { [month: number]: EventInfo } = {};
+    let eventInfo: EventInfo;
+    let result: ScatterData[] = [];
+    _.reduce(
+        scatterDataWithOpacity,
+        (eventInfoByMonth: { [month: string]: EventInfo }, item) => {
+            const month = Math.floor(item.x);
+            if (!(month in eventInfoByMonth)) {
+                // Provide initial value
+                eventInfo = {
+                    ...item,
+                    eventCount: 0,
+                    lastYInMonth: 0,
+                    censorCount: 0,
+                    lastRiskInMonth: item.atRisk!,
+                };
+            } else {
+                eventInfo = eventInfoByMonth[month];
+            }
+            eventInfoByMonth[month] = updateEventInfo(eventInfo, item);
+            return eventInfoByMonth;
+        },
+        eventInfoByMonth
+    );
+
+    result = _.map(eventInfoByMonth, (eventInfo, month) => {
+        const aggregatedScatter: ScatterData = {
+            x: parseInt(month),
+            y: eventInfo.lastYInMonth,
+            patientId: eventInfo.patientId,
+            uniquePatientKey: eventInfo.uniquePatientKey,
+            studyId: eventInfo.studyId,
+            status: eventInfo.status,
+            opacity: eventInfo.opacity,
+            group: eventInfo.group,
+            atRisk: eventInfo.lastRiskInMonth,
+            numberOfEvents: eventInfo.eventCount,
+            numberOfCensored: eventInfo.censorCount,
+        };
+        return aggregatedScatter;
+    });
+    return result;
+}
+
+function updateEventInfo(eventInfo: EventInfo, data: ScatterData): EventInfo {
+    let updatedEventInfo: EventInfo = eventInfo;
```

Comment:
keep in mind that you are not actually cloning this object here.  so any change you make below will actually apply to BOTH updatedEventInfo and eventInfo, since one is just a reference to the other.    This may or may not be a problem.  However, the code should not "pretend" that is producing something new.  I.e. if it's just mutating eventInfo input, then it should simply do that.

Reply:
Thanks, got it, I removed it.